### PR TITLE
Migrate HiveIntegrationSmokeTest to ConnectorTest

### PR DESCRIPTION
**Pull request by wendigo**
Fixes https://github.com/trinodb/trino/issues/7113

### CODE REVIEW

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveConnectorTest.java [hunk 1]
```diff
@@ -198,6 +195,35 @@ protected QueryRunner createQueryRunner()
...
+                return true;
+            default:
+                return super.hasBehavior(connectorBehavior);
+        }
+    }
+
+    @Override
+    public void testDelete()
+    {
+        // Hive connector supports row-by-row delete only for ACID tables
```
Comment (hashhar) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveConnectorTest.java:212]: SkipException?

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveConnectorTest.java [hunk 2]
```diff
@@ -198,6 +195,35 @@ protected QueryRunner createQueryRunner()
...
+        // but these currently cannot be used with file metastore.
+    }
+
+    @Test
+    @Override
+    public void testSortItemsReflectedInExplain()
```
Comment (hashhar) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveConnectorTest.java:218]: Regex should match from `super` if you declare `SUPPORTS_TOPN_PUSHDOWN: false`.

**Review by hashhar — APPROVED** [review]
Re-run CI with `tests:hive` label?

LGTM % comments and CI.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveConnectorTest.java (hunk 1 above)
Comment (wendigo) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveConnectorTest.java:213]: 👍🏼

**Comment by wendigo** [comment]
PTAL @hashhar

**Review by hashhar — APPROVED** [review]

**Pull request merged**